modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/fork/HandlerSuite.scala

```diff
@@ -0,0 +1,51 @@
+package org.tessellation.sdk.infrastructure.fork
+
+import org.tessellation.schema.generation.Generation
+import org.tessellation.schema.generators.peerIdGen
+import org.tessellation.schema.gossip._
+import org.tessellation.sdk.config.types.ForkInfoStorageConfig
+import org.tessellation.sdk.domain.fork.{ForkInfo, ForkInfoEntries, ForkInfoMap}
+import org.tessellation.sdk.infrastructure.fork.generators.genStoredForkInfoEntry
+
+import eu.timepit.refined.auto._
+import io.circe.syntax.EncoderOps
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object HandlerSuite extends SimpleIOSuite with Checkers {
+
+  test("received gossip gets added to the storage") {
+    val gen = for {
+      forkInfo <- genStoredForkInfoEntry.map(_._2)
+      peerId1 <- peerIdGen
+      peerId2 <- peerIdGen
+    } yield (forkInfo, peerId1, peerId2)
+
+    forall(gen) {
+      case (forkInfo, peerId1, peerId2) =>
+        for {
+          storage <- ForkInfoStorage.make(ForkInfoStorageConfig(10))
+          handler = forkInfoHandler.make(storage)
+          rawRumor = PeerRumorRaw(
+            peerId1,
+            Ordinal(Generation.MinValue, Counter.MinValue),
+            forkInfo.asJson,
+            ContentType.of[ForkInfo]
+          )
+
+          _ <- handler
+            .run((rawRumor, peerId2))
+            .getOrRaise(new Exception("failed to handle rumor"))
```

Comment:
Is this is correct?
Doesn't the fact that `RumorHandler.run` is a Kleisli on `OptionT` mean we have 3 states?

- F[Some(Unit)]
- F[None]
- F raised as error

It seems to me that in this case we're raising an error on the `None` case but from what I can tell by the implementation that (a) would never happen, and (b) isn't really an error since we're throwing the option away.

So isn the real error case when there's an error in `F` ?
Or put another way, do we just need this for the happy path:
```suggestion
          _ <- handler.run((rawRumor, peerId2)).value
```
and then have a test to show what is the error case?

Reply:
`None` is irrelevant and the `getOrRaise` is replaced by `value`.